src/digitalIdentity-frontend/src/app/services/error-message-service/error-message.service.ts

```diff
@@ -0,0 +1,81 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { InformationPopUpComponent } from 'src/app/shared/pop-up/information-pop-up/information-pop-up.component';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ErrorMessageService {
+
+  private dialogOpen = false
+
+  constructor(private matDialog: MatDialog) { }
+
+  openHttpErrorDialog(error: HttpErrorResponse) {
+
+    
+    let data = {
+      header: 'Process failed!',
+      text: 'Error ' + error.status + ' \n' + error.error,
+    };
+    
+    if (error.status == 401) {
+      data = {
+        header: 'Login session expired (401)',
+        text: 'Please log in to your account',
+      };
+    }
+    
+    if (error.status >= 500) {
+      data = {
+        header: 'Internal Error (' + error.status + ')',
+        text: 'The server isn\'t configured right, is unavailable at the moment or a connected service isn\'t working right. This error message could help:\n\n\n' + error.message + '\n\n' + error.error,
```

Comment:
'The server isn\'t configured correctly, is unavailable at the moment or a connected service doesn\'t work properly. ...'

would be better english :)

Reply:
I did it quickly, I hope that's okay :D